server/channels/app/import_functions_test.go

```diff
@@ -1667,48 +1667,76 @@ func TestImportImportUser(t *testing.T) {
 		assert.Equal(t, "", channelMember.ExplicitRoles)
 	})
 
-	// Test importing deleted guest with a valid team & valid channel name in apply mode.
-	username = model.NewUsername()
-	deleteAt = model.GetMillis()
-	deletedGuestData := &imports.UserImportData{
-		Username: &username,
-		DeleteAt: &deleteAt,
-		Email:    model.NewPointer(model.NewId() + "@example.com"),
-		Roles:    model.NewPointer("system_guest"),
-		Teams: &[]imports.UserTeamImportData{
-			{
-				Name:  &team.Name,
-				Roles: model.NewPointer("team_guest"),
-				Channels: &[]imports.UserChannelImportData{
-					{
-						Name:  &channel.Name,
-						Roles: model.NewPointer("channel_guest"),
+	t.Run("import deleted guest with a valid team & valid channel name in apply mode", func(t *testing.T) {
+		teamData := &imports.TeamImportData{
+			Name:            model.NewPointer(model.NewRandomTeamName()),
+			DisplayName:     model.NewPointer("Display Name"),
+			Type:            model.NewPointer("O"),
+			Description:     model.NewPointer("The team description."),
+			AllowOpenInvite: model.NewPointer(true),
+		}
+		appErr := th.App.importTeam(th.Context, teamData, false)
+		assert.Nil(t, appErr)
+
+		team, appErr2 := th.App.GetTeamByName(*teamData.Name)
+		require.Nil(t, appErr2, "Failed to get team from database.")
+
+		channelData := &imports.ChannelImportData{
+			Team:        teamData.Name,
+			Name:        model.NewPointer(NewTestId()),
+			DisplayName: model.NewPointer("Display Name"),
+			Type:        model.NewPointer(model.ChannelTypeOpen),
+			Header:      model.NewPointer("Channel Header"),
+			Purpose:     model.NewPointer("Channel Purpose"),
+		}
+		appErr2 = th.App.importChannel(th.Context, channelData, false)
+		assert.Nil(t, appErr2)
+		channel, appErr2 := th.App.GetChannelByName(th.Context, *channelData.Name, team.Id, false)
+		require.Nil(t, appErr2, "Failed to get channel from database")
+
+		username := model.NewUsername()
+		deleteAt := model.GetMillis()
```

Comment:
Does this need to be the current time or could it just be any static, positive value?

Reply:
Good point, I think it can be any static positive value, we probably don't have any check for `DeleteAt > CreateAt` but ideally it should conform to that condition.

Reply:
more fun with time.Now() :)